### PR TITLE
Introduce eks clusters and provide read-only example

### PR DESCRIPTION
**Pull request by ernestas2k**

### CODE REVIEW

File: castai/sdk/util_cluster.go [hunk 1]
```diff
@@ -0,0 +1,13 @@
+package sdk
+
+// Currently, sdk doesn't have generated constants for cluster status and agent status, declaring our own.
+const (
+	ClusterStatusReady    = "ready"
+	ClusterStatusDeleting = "deleting"
+	ClusterStatusDeleted  = "deleted"
+	ClusterStatusArchived = "archived"
+
+	ClusterAgentStatusOnline        = "online"
```
Comment (r0kas) [inline castai/sdk/util_cluster.go:10]: Maybe this should be `connected` As that's how we're showing in Console.

File: examples/eks_cluster_readonly/main.tf [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+  chart           = "castai-agent"
+  cleanup_on_fail = true
+
+  set {
+    name  = "provider"
+    value = "eks"
+  }
+  set_sensitive {
+    name  = "apiKey"
+    value = data.castai_agent_settings.agent_settings.token
```
Comment (r0kas) [inline examples/eks_cluster_readonly/main.tf:55]: Shouldn't this be set by calling backend to register new cluster?
What we should set is a token to call backend initially.

Comment (ernestas2k) [reply examples/eks_cluster_readonly/main.tf:55]: Currently, api.cast.ai provides the agent token and script before registering any cluster. As agreed, we're moving the token output into `castai_eks_cluster` for convenience and to address ease of supporting cluster-controller in the future.

**Review by r0kas — APPROVED** [review]

File: examples/external_cluster_eks/main.tf [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+  name = module.eks.cluster_id
+}
+
+### VPC
+
+module "vpc" {
+  source  = "terraform-aws-modules/vpc/aws"
+  version = "3.11.0"
+
+  name                 = "your-vpc"
```
Comment (matas-cast) [inline examples/external_cluster_eks/main.tf:41]: Should this be among variables?

**Pull request merged**